application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -183,20 +177,18 @@ private MessageCreateAction createModMessage(String reportReason,
             .setColor(AMBIENT_COLOR)
             .build();
 
-        MessageCreateAction message =
-                modMailAuditLog.sendMessageEmbeds(reportedMessageEmbed, reportReasonEmbed)
-                    .addActionRow(DiscordClientAction.Channels.GUILD_CHANNEL_MESSAGE.asLinkButton(
-                            "Go to Message", guild.getId(), reportedMessage.channelID,
-                            reportedMessage.id));
-
         Optional<Role> moderatorRole = guild.getRoles()
             .stream()
             .filter(role -> configModGroupPattern.test(role.getName()))
             .findFirst();
 
-        moderatorRole.ifPresent(role -> message.setContent(role.getAsMention()));
-
-        return message;
+        return guild.getTextChannelById(reportedMessage.channelID)
+            .retrieveMessageById(reportedMessage.id)
+            .map(Message::getJumpUrl)
+            .flatMap(messageUrl -> modMailAuditLog
+                .sendMessageEmbeds(reportedMessageEmbed, reportReasonEmbed)
+                .setContent(moderatorRole.map(Role::getAsMention).orElse(""))
+                .addActionRow(Button.link(messageUrl, "Go to Message")));
```

Comment:
When the message cant be found (retrieveMessageById fails), the whole thing fails and doesnt send the report anymore. Thats bad. Instead, it should maybe just leave away the button in that case.

Please test your edge cases more careful.

Reply:
if that'll happen the report message won't be filed, it is handled already

Reply:
> if that'll happen the report message won't be filed, it is handled already

but it should, thats my point. we want to have the report, even if the message ID cant be found anymore.

the old code also handled that gracefully by just having the button point to nowhere.

with ur code, the whole command just silently goes down at this stage.

Reply:
like here:

![modal](https://i.imgur.com/ij3QIOA.png)
![extraction](https://i.imgur.com/VapooRk.png)

you could just have the jump URL there. then u dont need that retrieve anymore and all those problems go away

Reply:
done